<Type Name="ErrorEventArgs" FullName="System.IO.ErrorEventArgs">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="8f2921fa15912959397b39597816d57dfaa68afe" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="it-IT" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
    <Meta Name="ms.locfileid" Value="30415845" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class ErrorEventArgs : EventArgs" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ErrorEventArgs extends System.EventArgs" />
  <TypeSignature Language="DocId" Value="T:System.IO.ErrorEventArgs" />
  <TypeSignature Language="VB.NET" Value="Public Class ErrorEventArgs&#xA;Inherits EventArgs" />
  <TypeSignature Language="C++ CLI" Value="public ref class ErrorEventArgs : EventArgs" />
  <AssemblyInfo>
    <AssemblyName>System.IO.FileSystem.Watcher</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.EventArgs</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary>Fornisce dati per l'evento <see cref="E:System.IO.FileSystemWatcher.Error" />.</summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <xref:System.IO.ErrorEventArgs> contiene il <xref:System.Exception> che ha causato il <xref:System.IO.FileSystemWatcher.Error?displayProperty=nameWithType> evento. Questa classe fornisce il <xref:System.IO.ErrorEventArgs.GetException%2A> metodo per recuperare l'eccezione.  
  
   
  
## Examples  
 Nell'esempio seguente crea una nuova istanza della <xref:System.IO.ErrorEventArgs> e la inizializza con un <xref:System.Exception>. Viene quindi chiamato <xref:System.IO.ErrorEventArgs.GetException%2A> per recuperare il <xref:System.Exception> e visualizzare il messaggio di errore. Non vi è alcun form associato a questo codice.  
  
 [!code-cpp[Classic ErrorEventArgs Example#1](~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic ErrorEventArgs Example/CPP/source.cpp#1)]
 [!code-csharp[Classic ErrorEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ErrorEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ErrorEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ErrorEventArgs Example/VB/source.vb#1)]  
  
 Nell'esempio seguente viene illustrato come creare un oggetto FileSystemWatcher per monitorare le modifiche ai file (Crea, Elimina, ridenominazioni e modifiche) che si verificano in un'unità disco. Nell'esempio viene inoltre illustrato come correttamente ricevere le notifiche di errore.  
  
 [!code-csharp[FileSystemWatcher#1](~/samples/snippets/csharp/VS_Snippets_CLR/FileSystemWatcher/cs/filesystemwatcher.cs#1)]
 [!code-vb[FileSystemWatcher#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/FileSystemWatcher/VB/FileSystemWatcher.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.IO.ErrorEventHandler" />
    <altmember cref="T:System.IO.FileSystemWatcher" />
    <altmember cref="M:System.IO.FileSystemWatcher.OnError(System.IO.ErrorEventArgs)" />
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ErrorEventArgs (Exception exception);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(class System.Exception exception) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.ErrorEventArgs.#ctor(System.Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ErrorEventArgs(Exception ^ exception);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.IO.FileSystem.Watcher</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="exception" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="exception">Oggetto <see cref="T:System.Exception" /> che rappresenta l’errore generato.</param>
        <summary>Inizializza una nuova istanza della classe <see cref="T:System.IO.ErrorEventArgs" />.</summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 Nell'esempio seguente crea una nuova istanza della <xref:System.IO.ErrorEventArgs> e la inizializza con un <xref:System.Exception>. Viene quindi chiamato <xref:System.IO.ErrorEventArgs.GetException%2A> per recuperare il <xref:System.Exception> e visualizzare il messaggio di errore. Non vi è alcun form associato a questo codice.  
  
 [!code-cpp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CPP/source.cpp#1)]
 [!code-csharp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="GetException">
      <MemberSignature Language="C#" Value="public virtual Exception GetException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig newslot virtual instance class System.Exception GetException() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.ErrorEventArgs.GetException" />
      <MemberSignature Language="VB.NET" Value="Public Overridable Function GetException () As Exception" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; virtual Exception ^ GetException();" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.IO.FileSystem.Watcher</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Exception</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>Recupera l'oggetto <see cref="T:System.Exception" /> che rappresenta l'errore generato.</summary>
        <returns>Oggetto <see cref="T:System.Exception" /> che rappresenta l’errore generato.</returns>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 Nell'esempio seguente crea una nuova istanza della <xref:System.IO.ErrorEventArgs> e la inizializza con un <xref:System.Exception>. Viene quindi chiamato <xref:System.IO.ErrorEventArgs.GetException%2A> per recuperare il <xref:System.Exception> e visualizzare il messaggio di errore. Non vi è alcun form associato a questo codice.  
  
 [!code-cpp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CPP/source.cpp#1)]
 [!code-csharp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>